ivy/functional/ivy/nest.py

```diff
@@ -345,9 +397,26 @@ def set_nest_at_indices(
     >>> print(nest)
     ivy.array([[1., 11., 3.], [4., 5., 22.]])
     """
+    nest_type = type(nest)
+    if shallow:
+        result = nest_type(nest)
+    else:
+        result = copy_nest(nest)
+    result = list(result) if isinstance(result, tuple) else result
     if not isinstance(values, (list, tuple)):
         values = [values] * len(indices)
-    [set_nest_at_index(nest, index, value) for index, value in zip(indices, values)]
+    for index, value in zip(indices, values):
+        if shallow:
+            result = set_nest_at_index(
+                nest, index, value, _result=result, shallow=shallow
+            )
+        else:
+            result = set_nest_at_index(
+                result, index, value, _result=result, shallow=shallow
+            )
```

Comment:
If we do that then `set_nest_at_index` won't receive updated nest in every loop in case of `shallow=False`.
`result = set_nest_at_index(nest, index, value, _result=result, shallow=shallow)` will only end up updating on the last index provided. Thats why I'm passing `result` as input in case of  `shallow=False`